core/src/main/scala/kafka/consumer/BaseConsumer.scala

```diff
@@ -91,6 +91,19 @@ class NewShinyConsumer(topic: Option[String], partitionId: Option[Int], offset:
     }
   }
 
+  private def seekToRealPositionsBeforeExit() {
+    val smallestUnconsumedOffsets = collection.mutable.Map[TopicPartition, Long]()
+    while (recordIter.hasNext) {
+      val record = recordIter.next()
+      val tp = new TopicPartition(record.topic(), record.partition())
```

Comment:
nit: no need for parenthesis on `topic` and `partition`.

Reply:
@hachikuji Sorry for the delay. I have no idea about how to write this test case. Writing a mock NewShinyConsumer to verify `seekToRealPositionsBeforeExit` is called when being shut down or a test case to consume some real records? If the latter is hit, could it be implemented from within `ConsoleConsumerTest.scala`? Please guide.

Reply:
Whatever is simplest. I had thought it might be easier to test `NewShinyConsumer` directly rather than going through the `ConsoleConsumer`, but if you feel it is easier to write a test in `ConsoleConsumerTest`, that is also fine.